### ISSUE TITLE
Bug: unwanted webpackPrefetch when using with webpackChunkName

### ISSUE THREAD

**Issue by Allain55** [webpack/webpack#12393]
# Bug report

**What is the current behavior?**
 Using webpackChunkName and webpackPrefetch together results in prefetching even when it's not wanted

**If the current behavior is a bug, please provide the steps to reproduce.**
Clone the repo: https://github.com/Allain55/webpack_issue_12393
$ npm i
$ npm start
Visit http://localhost:3500/pageA
Here "chunkA" is prefetched as it should be because webpackPrefetch is used in bundleA.js

Visit http://localhost:3500/pageB
Here "chunkA" is also prefetched but it shouldn't be as webpackPrefetch is not set in bundleB.js

**What is the expected behavior?**
 webpackChunkName should not affect prefetching

**Other relevant information:**
webpack version:  5.13
Node.js version: 15.5.1
Operating System: Ubuntu 16.04.7
Additional tools:

**Comment by alexander-akait** [comment]
Please describe the problem, not core and provide configuration or example of the problem

**Comment by Allain55** [comment]
OK I updated the issue

**Comment by alexander-akait** [comment]
Yep, bug

**Comment by akgupta0777** [comment]
I need more information about the issue. I didn't understand well.

**Comment by alexander-akait** [comment]
@akgupta0777 It is bug, do you want to fix it?

**Comment by akgupta0777** [comment]
Yes I want to fix it. But I have no idea what is the problem actually I ran the steps to reproduce the error but I need a proper view of the problem so that I can come up with a fix.

**Comment by alexander-akait** [comment]
@alexander-akait https://github.com/Allain55/webpack_issue_12393/blob/master/app/bundleB.js#L5, `webpackChunkName` should not set `webpackPrefetch` to `true`, I think it is not hard to fix, I think problem somewhere here https://github.com/webpack/webpack/blob/master/lib/dependencies/ImportParserPlugin.js, need debug

**Comment by akgupta0777** [comment]
@Allain55 Explain This

```
console.log('BundleB')

setTimeout(() => {
	import(
		
		'./chunkA.js'
		)
		.then(({default: chunkA}) => {
			chunkA();
		})
}, 1000);
```
I understand you are not prefetching chunkA in BundleB.js but you are importing it By doing this it will actually load chunkA.js and then you are attaching the chunkA() callback which will call after the promise is resolved.

**Comment by Allain55** [comment]
Hi @akgupta0777,
thanks for checking this. You're description is correct. This is the most simple example of dynamic importing I could come up with. In this case I'd need to use chunkA in some point but I don't want to prefetch it as it can wait

**Comment by akgupta0777** [comment]
@alexander-akait  what is groupOptions in line 98
https://github.com/webpack/webpack/blob/8ce54a045fb6edd7baf98a1b141ad3cd9d50ccad/lib/dependencies/ImportParserPlugin.js#L98

**Comment by akgupta0777** [comment]
Bump

**Comment by akgupta0777** [comment]
I came once again to this issue after researching on this webpack prefetch. So under the hood it actually just adds a
 `<link rel="prefetch" as="script" href="/AChunk.bundle.js">` in your HTML files that signals the browser to prefetch the module in idle state. 
So once browser prefetches the Module it stores that module in module cache so it will be loaded instantly when you want to load it.

in Page B dynamically importing ChunkA.js file returns a promise and then it will resolve and call the default export and chunk is console logged. 
ChunkA is not prefetched on Page B. Its loaded.

**Comment by akgupta0777** [comment]
@Allain55 You can clearly see here in HTML of Page A
you can see <link rel="prefetch" as="script" href="/AChunk.bundle.js"> in  head tag. which means it is prefetching.

![pageA](https://user-images.githubusercontent.com/51379307/120070707-821ee180-c0a9-11eb-9fb6-61aa8b7d9e5e.jpg)

but this is not in case of Page B as you can see below.
no link to prefetching it's natural promise resolved.

![PageB](https://user-images.githubusercontent.com/51379307/120070756-ada1cc00-c0a9-11eb-8333-6e83db252fd1.jpg)

**Comment by akgupta0777** [comment]
I don't think so it's kind of any bug.

**Comment by Allain55** [comment]
Hi @akgupta0777,
Did you follow the steps I described? I still get the same bug after cloning that repo. I tried with the latest webpack too and I still see that "/myChunk.bundle.js" is prefetched on pageB. 

In your picture a script named "AChunk.bundle.js" is prefetched on pageA but this should be "myChunk.bundle.js", there is no "Achunk.bundle.js"

**Comment by akgupta0777** [comment]
@Allain55 I changed the chunkname property to Achunk in bundleA and Bchunk in BundleB so that I can differentiate between the two and seperate bundles will be applied to seperate pages

**Comment by Allain55** [comment]
I see. If you use different webpackChunkNames or skip it than there is no bug. The bug only happens when webpackChunkName is set at least 2 times and webpackPrefetch is true at least 1 times but not 2 times. Since in this case we prefetch the same chunk I used the name "myChunk" for both but I only added webpackPrefetch for 1 chunk

**Comment by akgupta0777** [comment]
Ohh It means webpack chunknames have name collisions.

**Comment by akgupta0777** [comment]
But this doesn't make sense you actually importing same module dynamically twice in two different javascript files.

**Comment by Allain55** [comment]
That is why it is a chunk. I'd use the same chunk in 2 bundles eg. a gallery that is not needed on page load

**Comment by akgupta0777** [comment]
can you tell me what will happen if multiple webpages are prefetching the same resources ?. I mean they will be cached and when requesting on other pages will instantly serve the resource.

**Comment by Allain55** [comment]
Yes, the resources should be cached. But they should be cached regardless of prefetching.
I used prefetch to speed up the initial render of a big resource that is not needed right away.

**Comment by akgupta0777** [comment]
@Allain55 I mean that if for instance the pageA prefetches the chunk already and when user moved to pageB then another network call is made or not to prefetch the chunk again. Does this impacts performance ?

**Comment by Allain55** [comment]
Prefetching happens in the background with low priority so I don't think it has any noticeable performance impact. Still another network call will be made
